### PR TITLE
27 cupcakeselection full stack

### PR DESCRIPTION
**Pull request by TRossen89**

### CODE REVIEW

File: src/main/java/app/controllers/CartController.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import io.javalin.http.Context;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CartController {
+
+
+
+//TODO: Add these ctx.attributes every time cupcakeSelection.html is rendered
```
Comment (cph-dh223) [inline src/main/java/app/controllers/CartController.java:17]: Remove unused code

File: src/main/java/app/entities/Topping.java
Comment (cph-dh223) [inline src/main/java/app/entities/Topping.java:1]: See bottom comment

File: src/main/java/app/entities/Bottom.java
Comment (cph-dh223) [inline src/main/java/app/entities/Bottom.java:1]: Bottom and Topping can be 2 implementations of an abstract suberclass as they contain the exact same code

File: src/main/java/app/entities/Cart.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    private List<Orderline> orderlineList;
+
+
+    public Cart(List<Orderline> orderlineList) {
+        this.orderlineList = orderlineList;
+
+    }
+
+    public double getTotalPriceOfCart(){
```
Comment (cph-dh223) [inline src/main/java/app/entities/Cart.java:15]: The total value of an orderline is never calculated and is allways assumed to be known

File: src/main/java/app/persistence/OptionsMapper.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class OptionsMapper {
+
+    public static List<Topping> getAllToppings(ConnectionPool connectionPool) throws DatabaseException {
+
+        List<Topping> allToppings = new ArrayList<>();
+
+        String sql = "SELECT * FROM \"topping\"";
```
Comment (cph-dh223) [inline src/main/java/app/persistence/OptionsMapper.java:20]: unnessesary: 
``` Java
\"
```

File: src/main/java/app/persistence/OptionsMapper.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        return allToppings;
+
+    }
+
+    public static List<Bottom> getAllBottoms(ConnectionPool connectionPool) throws DatabaseException {
+
+        List<Bottom> allBottoms = new ArrayList<>();
+
+        String sql = "SELECT * FROM \"bottom\"";
```
Comment (cph-dh223) [inline src/main/java/app/persistence/OptionsMapper.java:45]: see other comment on this file

File: src/main/java/app/persistence/OrderMapper.java [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        double toppingPrice = 0;
+        double totalPrice;
+
+        String bottomName = "";
+        String toppingName = "";
+
+        Bottom newBottom = null;
+        Topping newTopping = null;
+
+        String sql = "SELECT * FROM \"bottom\" WHERE id=?";
```
Comment (cph-dh223) [inline src/main/java/app/persistence/OrderMapper.java:31]: 1. see options mapper comments
2. The select is over kill use something like
``` SQL
SELECT name, price
```

File: src/main/java/app/persistence/OrderMapper.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+                    newBottom = new Bottom(bottomId, bottomName, bottomPrice);
+                }
+            }
+        } catch (SQLException e) {
+            throw new DatabaseException("Something went wrong with the connection to the database");
+        }
+
+
+        String sql2 = "SELECT * FROM \"topping\" WHERE id=?";
```
Comment (cph-dh223) [inline src/main/java/app/persistence/OrderMapper.java:51]: see other comment in this file

File: src/main/java/app/persistence/OrderMapper.java [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+
+    private static void insertingOrderInDB(User currentUser, double totalPriceOfOrder, ConnectionPool connectionPool, List<Orderline> orderlineList) throws DatabaseException {
+
+        int generatedOrderId = 0;
+
+        long millis = System.currentTimeMillis();
+        java.sql.Date dateOfOrder = new java.sql.Date(millis);
+
+        String sql1 = "INSERT INTO \"orders\" (date, status, user_id, total_price) VALUES (?, ?, ?, ?)";
```
Comment (cph-dh223) [inline src/main/java/app/persistence/OrderMapper.java:106]: see options mapper comments

File: src/main/java/app/persistence/OrderMapper.java [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+                }
+            }
+        } catch (SQLException e) {
+            throw new DatabaseException("Error when inserting new order in db!");
+        }
+
+
+        for (Orderline orderline : orderlineList) {
+
+            String sql2 = "INSERT INTO \"orderlines\" (order_id, quantity, topping_id, bottom_id, total_price) VALUES (?, ?, ?, ?, ?)";
```
Comment (cph-dh223) [inline src/main/java/app/persistence/OrderMapper.java:132]: see options mapper comments

**Review by cph-dh223 — CHANGES_REQUESTED** [review]

File: src/main/java/app/controllers/CartController.java (hunk 1 above)
Comment (TRossen89) [reply src/main/java/app/controllers/CartController.java:17]: Lav en rendermetode, der returnerer lister af Bottoms og Toppings

**Review by cph-dh223 — APPROVED** [review]

**Pull request merged**